kakao/src/main/kotlin/com/agoda/kakao/configurator/DataInteractionDelegateBuilder.kt

```diff
@@ -0,0 +1,49 @@
+package com.agoda.kakao.configurator
+
+import android.support.test.espresso.DataInteraction
+import android.support.test.espresso.ViewAssertion
+import android.support.test.espresso.ViewInteraction
+import android.view.View
+import com.agoda.kakao.delegates.DataInteractionDelegate
+import com.agoda.kakao.delegates.ViewInteractionDelegate
+import com.agoda.kakao.delegates.factory.InteractionDelegatesFactory
+import org.hamcrest.Matcher
+
+class DataInteractionDelegateBuilder(
+    private val dataInteractionForBuilder: DataInteraction,
+    private val parentDataInteractionDelegate: DataInteractionDelegate?
+) {
+
+    internal val dataInteractionDelegate = object : DataInteractionDelegate {
+        override val dataInteraction: DataInteraction
+            get() = dataInteractionForBuilder
+
+        override fun onChildView(childMatcher: Matcher<View>): DataInteractionDelegate {
+            onChildView?.invoke(childMatcher)
```

Comment:
This approach prevents the delegation of the call to the real data interaction. Is this desired behavior?

Reply:
Why does it prevent? I have not understood.